src/map/unit.c

```diff
@@ -959,71 +964,60 @@ static enum unit_dir unit_getdir(const struct block_list *bl)
 	return ud->dir;
 }
 
-// Pushes a unit by given amount of cells into given direction. Only
-// map cell restrictions are respected.
-// flag:
-//  &1  Do not send position update packets.
-static int unit_blown(struct block_list *bl, int dx, int dy, int count, int flag)
+/**
+ * Pushes a unit by given amount of cells into given direction.
+ * @remark Only map cell restrictions are respected.
+ * @param bl: unit to push
+ * @param dir: direction of the push motion
+ * @param count: amount of cells to push
+ * @param no_update: don't send position update packets when true
+ * @return amount of pushed cells.
+ */
+static int unit_blown(struct block_list *bl, enum unit_dir dir, int count, bool no_update)
 {
-	if(count) {
-		struct map_session_data* sd;
-		struct skill_unit* su = NULL;
-		int nx, ny, result;
-
-		nullpo_ret(bl);
-
-		sd = BL_CAST(BL_PC, bl);
-		su = BL_CAST(BL_SKILL, bl);
-
-		result = path->blownpos(bl, bl->m, bl->x, bl->y, dx, dy, count);
-
-		nx = result>>16;
-		ny = result&0xffff;
+	if (count == 0)
+		return 0;
+	nullpo_ret(bl);
 
-		if(!su) {
-			unit->stop_walking(bl, STOPWALKING_FLAG_NONE);
-		}
+	int result = path->blownpos(bl, bl->m, bl->x, bl->y, dir, count);
+	int pushto_x = result >> 16;
+	int pushto_y = result & 0xffff;
```

Comment:
These variables are first used inside the `if (sd != NULL)` block, so please move them down there.